aggregator_core/src/datastore.rs

```diff
@@ -3608,17 +3608,14 @@ impl<C: Clock> Transaction<'_, C> {
                 "WITH batches AS (
                     SELECT
                         outstanding_batches.batch_id AS batch_id,
-                        COUNT(DISTINCT report_aggregations.client_report_id) AS count
+                        SUM(batch_aggregations.report_count) AS count
                     FROM outstanding_batches
-                    JOIN aggregation_jobs
-                      ON aggregation_jobs.task_id = (SELECT id FROM tasks WHERE task_id = $1)
-                     AND aggregation_jobs.batch_id = outstanding_batches.batch_id
-                    JOIN report_aggregations
-                      ON report_aggregations.aggregation_job_id = aggregation_jobs.id
-                     AND report_aggregations.state = 'FINISHED'
-                    GROUP BY outstanding_batches.batch_id
+                    JOIN batch_aggregations
+                      ON batch_aggregations.batch_identifier = outstanding_batches.batch_id
+                    WHERE outstanding_batches.task_id = (SELECT id FROM tasks WHERE task_id = $1)
+                    GROUP BY outstanding_batches.batch_id, batch_aggregations.aggregation_param
                 )
-                SELECT batch_id FROM batches WHERE count >= $2 LIMIT 1",
+                SELECT DISTINCT batch_id FROM batches WHERE count >= $2::bigint LIMIT 1",
```

Comment:
Q: why distinct (given `LIMIT 1`)? why explicitly cast to `bigint`?

Reply:
The bigint cast is necessary because otherwise the prepared query expects a `numeric` type, and there isn't a canned `ToSql` implementation for that.